pkg/cmd/auth/refresh/refresh.go

```diff
@@ -143,32 +156,63 @@ func refreshRun(opts *RefreshOptions) error {
 		return cmdutil.SilentError
 	}
 
-	var additionalScopes []string
-	if oldToken, _ := authCfg.Token(hostname); oldToken != "" {
-		if oldScopes, err := shared.GetScopes(opts.HttpClient, hostname, oldToken); err == nil {
-			for _, s := range strings.Split(oldScopes, ",") {
-				s = strings.TrimSpace(s)
-				if s != "" {
-					additionalScopes = append(additionalScopes, s)
+	credentialFlow := &shared.GitCredentialFlow{
+		Executable: opts.MainExecutable,
+		Prompter:   opts.Prompter,
+		GitClient:  opts.GitClient,
+	}
+
+	tmpScopes := credentialFlow.Scopes()
```

Comment:
Like the [other PR (https://github.com/cli/cli/pull/7597#discussion_r1240000033), I think that when using `https`, there should be parity between `login` and `refresh` but this change results in `workflow` scope being removed.

```bash
➜  cli git:(feature/6785) ✗ ~/workspace/cli/bin/gh auth status
github.com
  ✓ Logged in to github.com as williammartin (keyring)
  ✓ Git operations for github.com configured to use https protocol.
  ✓ Token: gho_************************************
  ✓ Token scopes: gist, read:org, repo, workflow
 
 ➜  cli git:(feature/6785) ✗ ~/workspace/cli/bin/gh auth refresh --reset-scopes
? Authenticate Git with your GitHub credentials? Yes
! Using secure storage could break installed extensions
! First copy your one-time code: ...
Press Enter to open github.com in your browser...
✓ Authentication complete.

➜  cli git:(feature/6785) ✗ ~/workspace/cli/bin/gh auth status
github.com
  ✓ Logged in to github.com as williammartin (keyring)
  ✓ Git operations for github.com configured to use https protocol.
  ✓ Token: gho_************************************
  ✓ Token scopes: gist, read:org, repo 
```